api/v1/model_handlers.go

```diff
@@ -0,0 +1,298 @@
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /node/ endpoint exposes metrics for the Node entity of the model.
+	ws.Route(ws.GET("/node/{node-name}/{metric-name}").
+		To(a.nodeMetrics).
+		Filter(compressionFilter).
+		Doc("export all node-level metrics").
+		Operation("nodeMetrics").
+		Param(ws.PathParameter("node-name", "The name of the node to lookup").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /namespace/ endpoint exposes aggregated metrics for the Namespace entity of the model.
+	ws.Route(ws.GET("/namespace/{namespace-name}/{metric-name}").
+		To(a.namespaceMetrics).
+		Filter(compressionFilter).
+		Doc("export all aggregated namespace-level metrics").
+		Operation("namespaceMetrics").
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /namespace/{}/pod endpoint exposes aggregated metrics for the Pod entity of the model.
+	ws.Route(ws.GET("/namespace/{namespace-name}/pod/{pod-name}/{metric-name}").
+		To(a.podMetrics).
+		Filter(compressionFilter).
+		Doc("export all aggregated pod-level metrics").
+		Operation("podMetrics").
+		Param(ws.PathParameter("namespace-name", "The name of the namespace to use").DataType("string")).
+		Param(ws.PathParameter("pod-name", "The name of the pod to use").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /namespace/{}/pod/{}/container endpoint exposes metrics for the Pod Container entity of the model.
+	ws.Route(ws.GET("/namespace/{namespace-name}/pod/{pod-name}/container/{container-name}/{metric-name}").
+		To(a.podContainerMetrics).
+		Filter(compressionFilter).
+		Doc("export a container-level metric for a pod container").
+		Operation("podContainerMetrics").
+		Param(ws.PathParameter("namespace-name", "The name of the namespace to use").DataType("string")).
+		Param(ws.PathParameter("pod-name", "The name of the pod to use").DataType("string")).
+		Param(ws.PathParameter("metric-name", "The name of the requested metric").DataType("string")).
+		Param(ws.QueryParameter("start", "Start time for requested metrics").DataType("string")).
+		Writes(MetricResult{}))
+
+	// The /node/{}/container endpoint exposes metrics for the Free Container entity of the model.
+	ws.Route(ws.GET("/node/{node-name}/container/{container-name}/{metric-name}").
```

Comment:
that this calls returns only free containers and not all containers is confusing. rename?